Remove duplication in .mlis, expose MarshalError, add docs

gaborigloi opened the pull request:
Fixes #87 

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

gaborigloi commented:
We should also deduplicate the implementations themselves at one point, but that will be much harder.

gaborigloi commented:
It would also be nice to upload the docs automatically somewhere.

mseri commented:
I agree on both comments :)

mseri commented on src/lib/idl.mli:
```diff
@@ -180,35 +161,33 @@ type server_implementation
-  val (@->) : 'a Param.t -> 'b fn -> ('a -> 'b) fn
-  val returning : 'a Param.t -> 'b Error.t -> ('a, 'b) comp fn
-  val declare : string -> string list -> 'a fn -> 'a res
+		Then we can the server itself can be obtained by passing the [implementation]
```
Little typo: ‘Then the server itself’...

mseri commented:
I think we could use the associated github pages for the documentation

mseri commented:
Tomorrow I’ll play with dune-release and do a deserved upstream release and announce

gaborigloi commented:
Thanks for the review, fixed the issues.
That sounds great, thanks!
I was wondering, should we split out more opam packages first, to avoid having a second breaking release afterwards? Because if we do #101 then we'll introduce the `angstrom` dependency (probably, if there isn't an ocamldoc parser already available), and then it'd be nice to add this extra dependency to a separate opam package, instead of making `rpclib` heavier.

mseri commented:
I am still not sure ablut parsing ocamldoc by hands though

mseri commented:
And I think it would be a minimal disruption anyways

mseri commented:
I was starting to set-up the  auto push but I don’t have enough privileges to setup github pages. @jonludlam can you increase my privileges ro do it?
